prospero-common/src/main/java/org/wildfly/prospero/actions/ApplyCandidateAction.java

```diff
@@ -706,4 +717,24 @@ private static void glold(Path installationFile, final Path target) throws Provi
             throw new ProvisioningException("Failed to persist " + target.getParent().resolve(target.getFileName() + Constants.DOT_GLOLD), e);
         }
     }
+
+    private Map<String, String> getChannelNameMap() {
+        final Path candidateFile = installationDir.resolve(ProsperoMetadataUtils.METADATA_DIR).resolve(ProsperoMetadataUtils.CANDIDATE);
+        Map<String, String> getInfo = new HashMap<String, String>();
+
+        try (BufferedReader br = new BufferedReader(new FileReader(candidateFile.toFile()))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                String[] parts = line.split("=");
+                if (parts.length == 2) {
+                    String dependencyName = parts[0];
+                    String dependencyVersion = parts[1];
+                    getInfo.put(dependencyName, dependencyVersion);
+                }
+            }
+        } catch (IOException e) {
+            System.err.println("An error occurred: " + e.getMessage());
```

Comment:
Could you use Log instead of System.err and provide some information about error - Unable to read the channel names file, etc.